FirebaseTestProject/Views/PostRow.swift

```diff
@@ -8,23 +8,79 @@
 import SwiftUI
 
 struct PostRow: View {
-    let post: Post
+    @Binding  var post: Post
+    @State var isPresenting: Bool = false
+    @State private var showAlert: Bool = false
+    @State private var systemImage: String = ""
+    let deletePostAction:((Post) -> Void)
+    
+    init(post: Binding<Post>, deletePostAction: @escaping (Post) -> Void) {
+        self._post = post
+        self.deletePostAction = deletePostAction
+    }
+    
     var body: some View {
-        VStack {
-            Text(post.title)
-                .font(.title)
-                .padding(.bottom, 8)
+        VStack(alignment: .leading, spacing: 0) {
+            HStack {
+                Text(post.title)
+                    .font(.largeTitle)
+                    .padding(.leading)
+                Spacer()
+                let userid = UserDefaults.standard.value(forKey: "userid") != nil ? UUID(uuidString: UserDefaults.standard.value(forKey: "userid") as! String) : UUID(uuidString: "00854E9E-8468-421D-8AA2-605D8E6C61D9")
+                if userid == post.authorid {
+                    Button(action: {
+                        showAlert = true
+                    }, label: {
+                        Label("", systemImage: "trash")
```

Comment:
I would replace the empty string with something more descriptive, like `"Delete"`. This makes the button accessible to users who rely on screen readers (e.g. VoiceOver) to use the app.

If you’d like only the icon to be visible, you can use the `IconOnlyLabelStyle` to make the "Delete" text invisible.

Reply:
Oh, nifty! I like that, thanks for this!